### PR TITLE
CI: Build major features without aio.

### PR DESCRIPTION
**Pull request by nihohit**
This will catch some compilation issues. For example, #1272,
and will warn about fields that are only used in async features.

### CODE REVIEW

File: redis/src/cluster.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ where
             connections: RefCell::new(HashMap::new()),
             slots: RefCell::new(SlotMap::new(cluster_params.read_from_replicas)),
             auto_reconnect: RefCell::new(true),
-            read_timeout: RefCell::new(None),
+            read_timeout: RefCell::new(Some(cluster_params.response_timeout)),
```
Comment (jaymell) [inline redis/src/cluster.rs:235]: Is this needed for this PR? If not can we move to a separate commit?

Comment (nihohit) [reply redis/src/cluster.rs:235]: I'll split the commits. This was added because the split build raised warnings about unused fields.

Comment (nihohit) [reply redis/src/cluster.rs:235]: done. in the long term, we should probably use https://github.com/TheBevyFlock/flag-frenzy, https://crates.io/crates/cargo-hack, or https://github.com/frewsxcv/cargo-all-features to properly test this.

Comment (jaymell) [reply redis/src/cluster.rs:241]: Ah, found it :laughing: Should this be re-based on the other PR?

Comment (nihohit) [reply redis/src/cluster.rs:241]: actually, this PR can be closed in favor of the other PR, since it gives a more through check.
